src/BlazorUI/Bit.BlazorUI.Tests/SnackBar/BitSnackBarTests.cs

```diff
@@ -0,0 +1,205 @@
+            parameters =>
+            {
+                parameters.Add(p => p.AutoDismiss, autoDismiss);
+            }
+        );
+
+        await com.Instance.Show("title");
+
+        //Added a sec delay to be sure we are asserting after AutoDismissTime passed
+        await Task.Delay(com.Instance.AutoDismissTime + TimeSpan.FromSeconds(1));
+
+        var items = com.FindAll(".bit-snb-itm");
+
+        Assert.AreEqual(autoDismiss ? 0 : 1, items.Count);
+    }
+
+    [DataTestMethod,
+        DataRow(BitSnackBarType.Info),
+        DataRow(BitSnackBarType.Warning),
+        DataRow(BitSnackBarType.Success),
+        DataRow(BitSnackBarType.Error),
+        DataRow(BitSnackBarType.SevereWarning),
+        DataRow(null)
+    ]
+    [TestMethod]
+    public async Task BitSnackBarTypeTest(BitSnackBarType? type)
+    {
+        var com = RenderComponent<BitSnackBar>();
+
+        if (type.HasValue)
+            await com.Instance.Show("title", string.Empty, type);
+        else
+            await com.Instance.Show("title");
+
+        var element = com.Find(".bit-snb-itm");
+
+        var typeClass = type switch
+        {
+            BitSnackBarType.Info => $"bit-snb-info",
+            BitSnackBarType.Warning => $"bit-snb-warning",
+            BitSnackBarType.Success => $"bit-snb-success",
+            BitSnackBarType.Error => $"bit-snb-error",
+            BitSnackBarType.SevereWarning => $"bit-snb-severe-warning",
+            _ => "bit-snb-info"
+        };
+
+        Assert.IsTrue(element.ClassList.Contains(typeClass));
+    }
+
+    [TestMethod]
+    public async Task BitSnackBarCloseButtonTest()
+    {
+        var com = RenderComponent<BitSnackBar>();
+
+        await com.Instance.Show("title");
+
+        var closeButton = com.Find(".bit-snb-cbt");
+        closeButton.Click();
+
+        var items = com.FindAll(".bit-snb-itm");
+
+        Assert.AreEqual(0, items.Count);
+    }
+
+    [TestMethod]
+    public async Task BitSnackBarDismissIconNameTest()
+    {
+        var com = RenderComponent<BitSnackBar>(
+            parameters =>
+            {
+                parameters.Add(x => x.DismissIconName, "Go");
+            }
+        );
+
+        await com.Instance.Show("title");
+
+        var closeButtonIcon = com.Find(".bit-snb-cbt > .bit-icon");
+
+        Assert.IsTrue(closeButtonIcon.ClassList.Contains("bit-icon--Go"));
+    }
+
+    [TestMethod]
+    public async Task BitSnackBarTitleTemplateTest()
+    {
+        var wrappedCom = RenderComponent<BitSnackBarWithTitleTemplate>();
+
+        var com = wrappedCom.FindComponent<BitSnackBar>();
+
+        await com.Instance.Show("title");
+
+        var titleTemplateElements = com.FindAll(".bit-snb-hdr > *").ToList();
+        //remove button
+        titleTemplateElements.RemoveAt(0);
+
+        Assert.AreEqual("<span>title</span><span>second element</span>", string.Join(string.Empty, titleTemplateElements.Select(x => x.OuterHtml).ToArray()));
```

Comment:
why not use the InnerHtml of the parent instead of combining the OuterHtml of children?

Reply:
The problem here is we have a Button here that is in the same level of Template content.
We should able to get those, I tried to make it possible with correct selector but I couldn't do that, so I removed it like that.

Reply:
check the bUnit docs especially this page to use a better approach for these tests:

https://bunit.dev/docs/verification/verify-component-state.html

Reply:
Thanks a lot, I'll revise my code after read this